MapboxMobileEvents/Categories/NSUserDefaults+MMEConfiguration.m

```diff
@@ -312,7 +318,21 @@ - (NSString *)mme_legacyUserAgentString {
 // MARK: - Location Collection
 
 - (BOOL)mme_isCollectionEnabled {
-    return ![self boolForKey:MMECollectionDisabled];
+    BOOL collectionEnabled = ![self boolForKey:MMECollectionDisabled];
+    
+#if TARGET_OS_SIMULATOR
+    // disable collection in the smulator unless explicitly enabled for testing
+    if (!self.mme_isCollectionEnabledInSimulator) {
+        collectionEnabled = NO;
+    }
+#endif
+
+    // if not explicitly disalbed, or in simulator, check for low power mode
+    if (collectionEnabled && [NSProcessInfo instancesRespondToSelector:@selector(isLowPowerModeEnabled)]) {
+        collectionEnabled = !NSProcessInfo.processInfo.isLowPowerModeEnabled;
```

Comment:
We may need to bring back some of the previous code to keep some warnings silent for the Maps folks. iOS 8 is still supported by us (barely), we may need to keep in a OS check as well.
```
// Only check power mode if compiling with the iOS 9+ SDK and, at runtime, if the API exists
#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 90000
    if ([NSProcessInfo instancesRespondToSelector:@selector(isLowPowerModeEnabled)]) {
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wpartial-availability"
        isPowerModeCompatibleWithCollection = ![[NSProcessInfo processInfo] isLowPowerModeEnabled];
#pragma clang diagnostic pop
    }
#endif
```

Reply:
I was expecting to get a warning on that, and double checked the targets. I think the `instancesRespondToSelector` may be sufficient to squash the warning. I would setup a simulator to check, but the oldest one available is iOS 10!

Reply:
Whoops, forgot: CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE